File: backend/src/main/java/com/zhiyu/entity/AuditLog.java
package com.zhiyu.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 审计日志表（PRD 8.12）
 * 不含逻辑删除字段，日志只增不改不删
 */
@Data
@TableName("audit_log")
public class AuditLog {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long operatorId;

    private Integer operatorRole;

    private String action;

    private String targetType;

    private Long targetId;

    private String beforeJson;

    private String afterJson;

    private String ipAddress;

    private LocalDateTime createdAt;
}

File: front/src/admin/api/auditLog.ts
/**
 * 审计日志 API 模块
 * 对应后端 AuditLogController：/api/v1/admin/audit-logs
 */

import http from './http'
import type { ApiResult } from '../types'

/** 审计日志分页项（后端 AuditLogVO） */
export interface AuditLogItem {
  id: number
  operatorId: number
  operatorName: string
  operatorRole: number
  action: string
  targetType: string
  targetId: number | null
  beforeJson: string | null
  afterJson: string | null
  ipAddress: string | null
  createdAt: string
}

/** 审计日志分页响应 */
export interface AuditLogPage {
  list: AuditLogItem[]
  total: number
}

/** 分页查询审计日志 */
export async function listAuditLogs(
  pageNum: number,
  pageSize: number,
  action?: string,
): Promise<AuditLogPage> {
  const { data } = await http.get<ApiResult<AuditLogPage>>(
    '/api/v1/admin/audit-logs',
    { params: { pageNum, pageSize, ...(action ? { action } : {}) } },
  )
  return data.data
}